NameGen - Spice up the station names

zonkmachine opened the pull request:
* Add separate name formats for asteroids
    * Occasional Roman numerals to planet names
    * Add new station names/functions
    * Add occasional random station number

What planet name isn't improved with a Roman numeral?
![namegenroman1](https://github.com/user-attachments/assets/2064dc44-6c4d-4c0d-8a8b-a59e37a7701a)

Numbers to spice up station names
![namegennumbers2](https://github.com/user-attachments/assets/7e4e3997-39bd-48e4-85e0-952ac72f7c29)

The smaller bodies gets their own set of name formats
![namegencoredrilling](https://github.com/user-attachments/assets/496d1375-e6cf-4226-9a5d-c05bba8e18f1)

Can there be more than one instance of the same number. Yes. Same name and number? Not likely, but yes. This is what I could come up with in a short amount of time that made a reasonable improvement in the naming of bodies. I think this could be improved further if the pioneer universe was populated with companies/entities that placed their assets, ships and stations, in the systems and that had it's own decals, naming schemes etc.
Core Drilling Station? Yes, they are an efficient way to process asteroids... ;)

@impaktor I saw https://github.com/impaktor/pioneer/commit/a628fcf7a43254eacf7c8cd8f1161d8653e528fc and was actually trying something similar but I'm not happy with the results yet. 

Addresses https://github.com/pioneerspacesim/pioneer/issues/897

impaktor reviewed (commented):
Left some comments. In addition:

Since I just finished Mass Effect (1-2), I'm thinking you could also add "citadel" and "ward" to the station names.

zonkmachine commented on data/libs/NameGen.lua:
```diff
@@ -120,36 +120,64 @@ NameGen = {
 
+			-- Occasional Roman numeral stuck to a planet name looks good
+			local suffix = ""
+			local srand = rand:Integer(0, 999)
```
OK. Didn't think about that. I drop it for now. I'm trying letters in front of instead...

zonkmachine replied:
Actually, the order of the planets are already indicated by lower case letters.

zonkmachine commented:
How about occasional Letters in front of?

![namegenprefix2](https://github.com/user-attachments/assets/62c36499-9aee-458f-ab93-f972249dcdfd)
![namegenprefix4](https://github.com/user-attachments/assets/abdbd4dc-6c5d-49c4-bf16-1d5c45282eb8)
![namegenprefix3](https://github.com/user-attachments/assets/2294c9f0-bcd5-406c-8db0-361a8feeee51)

zonkmachine commented on data/libs/NameGen.lua:
```diff
@@ -159,7 +172,9 @@ NameGen.outdoorPlanetFormats = {
 }
 
 NameGen.rockPlanetFormats = {
-	"{name}'s Mine",
```
I moved this out of the planet names and added mines as station name instead.

impaktor commented:
Thought: High roman numerals are "nicer". 
- Pope Pius XII looks cooler than Pope Pius I. 
- current King of Sweden is Carl XVI, but he's not the 16th "{K,C}arl", since they started counting on [IX](https://en.wikipedia.org/wiki/Charles_IX_of_Sweden), 500 years ago, to give illusion of a long noble line, he's just the 7th (I think), at least that is well documented. 

(No action needed, except pondering)

zonkmachine commented:
I'm starting to swing back to Roman numerals. It is a bit random, yes,  but It looks good and the planets are already distinguished by lower case letters so I don't think that interferes.

bszlrd commented:
How about some very generic corporate names for Haber systems? (Can it be even selected by faction)
Like Commercial Port B-24
Mining Installation DZ-93B

zonkmachine commented:
> How about some very generic corporate names for Haber systems? (Can it be even selected by faction) Like Commercial Port B-24 Mining Installation DZ-93B

I think so. Pioneer jumps back and forth between lua and c++ when building the star sysem, but stations are created right after factions are done. I tested to print the faction name from NameGen.lua and it works. I'll have a go.

zonkmachine commented:
@bszlrd This is all that's needed to get access to the faction.
`if body.system.faction.name == "Haber Corporation" then`

Can you be more specific on what you had in mind?

bszlrd commented:
I was thinking about having a different naming scheme just for Haber. Kinda like how cultures are handled, but for systems and their elements.

zonkmachine commented:
Yes, I understand. I'm low on ideas of how that naming scheme would look like.

bszlrd commented:
I'll whip up something. What are the elements/parts? How names are assembled?

Web-eWorks commented:
Roman numerals should only be used on planets which are getting a "custom" name and do not follow the nomenclature of the existing system name (i.e. "Dolisium A c" should not have an additional numeral suffix, but "Wolfgargoyle IX" is fine).

zonkmachine commented:
> I'll whip up something. What are the elements/parts? How names are assembled?

Yes.  Numbers/Letters, how many and where? Only a select part of the alphabet, hexadecimals? Stuff like that.

impaktor commented:
> Only a select part of the alphabet, hexadecimals? Stuff like that.

Now there's a thought: each planet named after a commit hash of the project itself... :wink:

zonkmachine commented:
Added back an improved version of the Roman numerals

zonkmachine commented on data/libs/NameGen.lua:
```diff
@@ -120,36 +121,56 @@ NameGen = {
 
+		-- Occasional Roman numeral stuck to a planet name looks good
+		local suffix = ""
+		if rand:Number() < 0.4 then
```
0.4 may be a bit high. Leaving it like this for now.

bszlrd commented:
How about this:
Names are kinda like URLs:
I might have overthunk a bit, but how about this?
Names would work kinda like an URL:
> Watt-32E (system)
> Gr.Proc.Kstovo.3.Watt-32E (settlement)
> Or.Inst.Pullman.2.Egerton-15B (settlement)
> 2.UN.Clement-8C (uninhabited planet)
> 2.IN.Egerton-15B (inhabited)

Structured like this: 
> <num>Type.Role.Name.number.System
Where:
<num> Number of the body in the system (as with others, 1,2,3  4a 4b etc)
Type: Orbital (Or) or Ground (Gr)
Role: Abbreviated function of the settlement (Inst = Installation, Proc = Processing
Name: Name of the settlement - from a list of historical company towns
Number: a random number 1-99 for abscureness
System: the system the city is.

I've assembled a list of names that could be used:

<details>
<summary>System names: Prominent people of the industrial revolution. Industrialists, inventors, engineers</summary>

- Symthe - founder of the British East India company
- Boulton - partner of Watt
- Watt - inventor
- Roebuck - industrialist, Watt's first partner
- Lombe - a first to use powered machinery in silk throwing in england
- Kay - inventor of the flying shuttle for textile making
- Hargreaves  - spinning jenny
- Arkwright - inventor
- Crompton - inventor
- Cartwright - inventor
- Roberts - inventor
- Whitney - inventor
- Darby - inventor
- Clerke - enterpreneur
- Wilkinson - industrialist
- Smeaton - engineer, civin infrastructure engineer
- Cort - inventor
- Hall - inventor
- Neilson - inventor
- Huntsman - inventor
- Savery - inventor, engineer
- Newcomen
- Trevithick - inventor, engineer
- Evans - inventor, engineer
- Fox - toolmaker
- Murray - toolmaker
- Roberts - engineer
- Clement - engineer, insdustrialist
- Whitworth - engineer, inventor
- Leblanc - chemist, inventor
- Tennant - chemist, industrialist
- Berthollet - chemist
- Aspdin - chemist inventor
- Brunel - engineer
- Murdoch - chemist, engineer
- Chance - glassmaker brothers
- Tull - agriculturalist
- Meikle - engineer
- Davy
- Stephenson
- Egerton
- Metcalf - road builder
- Telford - engineer
- McAdam - civil engineer
- Locke
- Wedgewood - potter, enterpreneur
- Cockerill - industrialist
- Somers - inventor
- Slater - industrialist
- Lowell - businessman
- Blanchard - inventor
- Bessemer - inventor
</details>

<details> 
<summary> Settlement names: Historical corporate towns and cities from around the world </summary>

- Pullman
- Bournville
- Creswell
- Saltaire
- Copley
- Creswell
- Bolsover
- Güell
- Fordlândia
- Arvida
- Batawa
- Humberstone
- Sewell
- Kuusankoski
- Le Creusot
- Leverkusen
- Ludwigshafen
- Limburgerhof
- Neuölsburg
- Wolfsburg
- Salzgitter
- Eisenhüttenstadt
- Swords
- Oranjemund
- Widzew
- Svit
- Kiruna
- Skoghall
- Pripyat
- Segundo
- Marktown
- Scotia
- Cass
- Bay Lake
- Zlín
- Sezimovo Ústí
- Billund
- Bjerringbro
- Nordborg
- Struer
- Noisiel
- Hayange
- Thumeries
- Sindelfingen
- Schwedt
- Hoyerswerda
- Dunaújváros
- Tiszaújváros
- Petőfibánya
- Rochfortbridge
- Rosignano
- Colleferro
- Dnestrovsc
- Batadorp
- Heveadorp
- Philipsdopr
- Barentsburg
- Grumant
- Longyearbyen
- Pyramiden
- Sveagruva
- Rjukan
- Nowa Huta
- Giszowiec
- Kawęczyn
- Sieverdonetsk
- Magnitogorsk
- Norilsk
- Monchegorsk
- Zapolyarny
- Nikel
- Kovdor
- Olenegorsk
- Apatity
- Kirovsk
- Surgut
- Nizhevartovsk
- Nefteyugansk
- Kstovo
- Kirishi
- Ivanovo
</details>

<details>
<summary> Type: I made a relatively short list of types </summary>

- Mine
- Plnt - plant
- Cmrc - Commercial
- Prcm - Procurement
- Proc - Processing
- Buro - Bureaucratic
- Inst - Installation
- Lab - Laboratory
- Site 
- Shp - Shipyard
- Slvg - Salvage yard
- Maint - Maintenance yard
- Obs - Observation
- Sec - Security

Orbitals:
- Proc
- Plnt
- Cmrc
- Prcm
- Buro
- Inst
- Lab
- Site
- Shp
- Slvg
- Maint
- Obs
- Sec
</details>

zonkmachine commented:
You could also give the factions the option to have a list of famous people to choose from and the culture name files could have lists of names of places and nature objects to use in naming of places. It's maybe a bit out of scoop for this PR though?
For instance `Watt-32E (system)`, the system names are generated on the c++ side.
I will give this some thought and in the mean time I'll clear this PR up.

zonkmachine commented:
Here's the current spread of the numerals with `local srand = 1 + math.floor((rand:Number() ^ 3) * #romanNumerals)`
_Message to self:_ `sort < filename | uniq -c`

652 II
177 III
109 IV
 98 V
 75 VI
 55 VII
 60 VIII
 63 IX
 58 X
 38 XI
 51 XII
 50 XIII
 37 XIV
 35 XV
 26 XVI
 41 XVII
 21 XVIII
 25 XIX

You wouldn't use '**I**' for the first version of something, right? I omitted it from the numeral list.

zonkmachine commented:
OK. Squashed and rebased on master. I think the balance is pretty good right now.

![namegenbase4](https://github.com/user-attachments/assets/b3edc41d-d2d9-49ae-ab16-e6e1917d115c)

zonkmachine commented:
I think this can be merged.

Web-eWorks commented:
I'm a little hesitant with the addition of numeral suffixes as it can give the impression of an "amnesiac connection" - i.e. the game implies to the player that a Solvang I exists somewhere and was important enough that there's a second Solvang (and thus important enough for a player to try to find it to discover some hidden lore). I won't object to the merging of this PR over it, however.

I'd suggest it might be worth considering flipping the order of some names for greater variety, and adding one or two more prefix/suffix schemes.

For example:
`6013a Oberon` - Prefix, uses a random number 1000-9999 followed by a single letter a-x, mimics the IAU naming scheme for asteroids (does not have to be restricted to asteroids!).
`Core Drilling Station Titania-IV` - Flip the order of the name and the descriptor, optionally add a numeric suffix.
`HL54 Ross` - Prefix, uses two random numbers and two random letters before the body name.

I'm definitely happy to see some work done in expanding the quality and breadth of the Pioneer universe. Thank you very much for this PR!

impaktor commented:
> I'm a little hesitant with the addition of roman numeral suffixes ... game implies to the player that a Solvang I exists somewhere and was important enough that there's a second Solvang 

I'm fine with this especially for stations. Basically, there's an F-18, F-22, F-35, but intermediate were only design prototypes / defunct. (I remember the hype around F-19 you could buy models of it, there was the Microprose game)

For celestial bodies it might be weird, as you don't plan / design or iterate them, as you could do with stations.

Web-eWorks commented on data/libs/NameGen.lua:
```diff
@@ -195,11 +224,20 @@ NameGen.orbitalStarportFormats = {
-	"{name}",
+	"{name}{number}",
+	"{name}{number}",
+	"{name}{number}",
```
Note that this produces an exceptionally high number of single-word starport names.

![image](https://github.com/user-attachments/assets/e76f32ab-a59e-4404-927c-d985890021fa)
(Taiton is not the direct result of random generation, all other names are as-generated.)

This isn't a problem with long multi-syllable names, but a station named *only* "Lloyd" (or even "Rothmann") is quite immersion-breaking. I will most likely address this in a follow-on PR - I'm happy with the additional diversity of suffixes this PR adds and it's been sitting open long enough already.

zonkmachine replied:
Hm, **Taiton**? I don't see the same list of stations for NN 3737.

> a station named _only_ "Lloyd" (or even "Rothmann") is quite immersion-breaking.

I think rather the opposite. Too many station names with a name and a random 'Spaceport' stuck onto it is more immersion breaking for me. Details like that isn't of great importance to me though. Thanks for the merge!

> I will most likely address this in a follow-on PR

Please tag me... :)

Web-eWorks replied:
> Hm, **Taiton**? I don't see the same list of stations for NN 3737.

Yeah, I clicked randomize a bunch in the system editor to get an idea of what the distribution of names was.

> I think rather the opposite. Too many station names with a name and a random 'Spaceport' stuck onto it is more immersion breaking for me. Details like that isn't of great importance to me though. Thanks for the merge!
> 
> Please tag me... :)

The thing here that's immersion breaking (at least in English speaking / British/American style countries) is that towns/cities can have a single "person name" as their name, but transit infrastructure (airports, metro stations, etc) are almost always named e.g. `<Area name> Airport` or `<Area name> Station`. For example, it's "John F. Kennedy International Airport" or "Dallas-Fort Worth Intl. Airport", not "John F. Kennedy" or "Dallas-Fort Worth".

Given that these names are assigned to the spaceport specifically (an expectation set by the icon) there's an "uncanny valley" feeling when looking at those names, especially next to a name that does follow convention.

Perhaps it might help your sense of immersion if we were to generate some `<First Name> <Last Name> Spaceport` style name to reduce the deluge of `<Name> Spaceport`s?

Web-eWorks commented:
Thanks for your work on this @zonkmachine!
